File: app/channels/chat_channel.rb
class ChatChannel < ApplicationCable::Channel
  def subscribed
    stream_from "chat_channel_#{params[:booking_id]}" if params[:booking_id].present?
  end

  # def unsubscribed
  #   # Any cleanup needed when channel is unsubscribed
  # end

  # def speak(data)
  #   ActionCable.server.broadcast "room_channel", message: data['message']
  # end

end

File: app/javascript/channels/chat_channel.js
import consumer from "./consumer"

consumer.subscriptions.create("ChatChannel", {
  connected() {
    var booking_id = document.getElementById("booking_id")
    // Called when the subscription is ready for use on the server
  },

  disconnected() {
    // Called when the subscription has been terminated by the server
  },

  received(data) {
    // Called when there's incoming data on the websocket for this channel
  }
});
